store/src/db/mod.rs

```diff
@@ -0,0 +1,536 @@
+                }
+            })
+            .await
+            .map_err(|_| anyhow!("Get block header task failed with a panic"))?
+    }
+
+    /// Loads all the block headers from the DB.
+    pub async fn get_block_headers(&self) -> Result<Vec<BlockHeader>, anyhow::Error> {
+        self.pool
+            .get()
+            .await?
+            .interact(|conn| {
+                let mut stmt = conn.prepare("SELECT block_header FROM block_headers;")?;
+                let mut rows = stmt.query([])?;
+                let mut result = vec![];
+                while let Some(row) = rows.next()? {
+                    let block_header_data = row.get_ref(0)?.as_blob()?;
+                    let block_header = BlockHeader::decode(block_header_data)?;
+                    result.push(block_header);
+                }
+
+                Ok(result)
+            })
+            .await
+            .map_err(|_| anyhow!("Get block headers task failed with a panic"))?
+    }
+
+    /// Save a [Note] to the DB.
+    #[cfg(test)]
+    pub async fn add_note(
+        &self,
+        note: Note,
+    ) -> Result<usize, anyhow::Error> {
+        let num_rows = self
+            .pool
+            .get()
+            .await?
+            .interact(move |conn| -> Result<_, anyhow::Error> {
+                use miden_node_store::errors::DbError;
+
+                let mut stmt = conn.prepare(
+                    "
+                    INSERT INTO
+                    notes
+                    (
+                        block_num,
+                        note_index,
+                        note_hash,
+                        sender,
+                        tag,
+                        num_assets,
+                        merkle_path
+                    )
+                    VALUES
+                    (
+                        ?1, ?2, ?3, ?4, ?5, ?6, ?7
+                    );",
+                )?;
+                let res = stmt.execute(params![
+                    note.block_num,
+                    note.note_index,
+                    note.note_hash.ok_or(DbError::NoteMissingHash)?.encode_to_vec(),
+                    note.sender,
+                    note.tag,
+                    note.num_assets,
+                    note.merkle_path.ok_or(DbError::NoteMissingMerklePath)?.encode_to_vec(),
+                ])?;
+                Ok(res)
+            })
+            .await
+            .map_err(|_| anyhow!("Add note task failed with a panic"))??;
+
+        Ok(num_rows)
+    }
+
+    /// Searchs a block after `block_num` which contains note(s) matching [tag]s, and returns all
+    /// matching notes.
+    ///
+    /// # Returns
+    ///
+    /// An empty vector if the blocks after `block_num` don't contains notes matching `tags`.
+    /// Otherwise the matching notes from the next block are returned. Note that this method returns
+    /// notes from a single block.
+    pub async fn get_notes_since_block_by_tag_and_sender(
```

Comment:
nit: I'd probably mention `sender` in the comments here.

Reply:
changed